src/export/sketch-exporter.ts

```diff
@@ -21,6 +28,37 @@ export class SketchExporter {
 			getRectangleName: () => 'background',
 			skipSystemFonts: true
 		});
-		FileExtraUtils.writeFileSync(path, JSON.stringify(sketchPage.toJSON(), null, '\t'));
+
+		return JSON.stringify(sketchPage.toJSON(), null, '\t');
+	}
+
+	// tslint:disable-next-line promise-function-async
+	public createExport(webview: WebviewTag): Promise<ExportResult> {
+		return new Promise((resolve, reject) => {
+			// tslint:disable-next-line no-any
+			ipcRenderer.on('export-as-sketch-done', (_: Event, payload: any) => {
```

Comment:
While we could cast the payloads to some well-defined interface I'd argue we should not – any other part of the application could break the event handlers then. 

I pushed another approach that moves the message type refinements to dedicated helper functions using the `is` type operator

Reply:
See https://github.com/meetalva/alva/pull/388/commits/5c49524cd3d79ecdcfd8c4532e0679889f612436#diff-687d49822532c99f78577a641d66ac93R74 and https://github.com/meetalva/alva/pull/388/commits/5c49524cd3d79ecdcfd8c4532e0679889f612436#diff-687d49822532c99f78577a641d66ac93R91

Reply:
To reiterate on this - what I want to avoid here are real, user-facing bugs that are very common in JavaScript application. 

I think that the edges of the system (e.g. message handlers) have to jump through all the hoops we know from those, even when writing the thing in TypeScript.

```
ipcRenderer.on('export-as-sketch-done', (payload: any) => {
  const message = payload as SomeInterface; 
  message.type // may produce a user-facing bug, e.g. for message below
})

```

suggets a type safety that is not there. Nothing (including `tsc`) prevents anyone from doing this – afaik there is no straightforward way to even warn about possible errors during development time.

```
emitter.send('export-as-sketch-done', 'thing');
```

Reply:
Bringing in the rest of the time is a good idea 👍